p2p/kademlia/hashtable.go

```diff
@@ -2,155 +2,205 @@ package kademlia
 
 import (
 	"bytes"
-	"context"
 	"math"
-	"math/big"
 	"math/rand"
-	"net"
 	"sort"
 	"sync"
 	"time"
-
-	"github.com/pastelnetwork/gonode/common/errors"
 )
 
 const (
-	iterateStore = iota
-	iterateFindNode
-	iterateFindValue
+	// IterateStore - iterative store the data
+	IterateStore = iota
+	// IterateFindNode - iterative find node
+	IterateFindNode
+	// IterateFindValue - iterative find value
+	IterateFindValue
 )
 
 const (
-	// a small number representing the degree of parallelism in network calls
-	alpha = 3
+	// Alpha - a small number representing the degree of parallelism in network calls
+	Alpha = 3
 
-	// the size in bits of the keys used to identify nodes and store and
+	// B - the size in bits of the keys used to identify nodes and store and
 	// retrieve data; in basic Kademlia this is 160, the length of a SHA1
-	b = 160
+	B = 160
 
-	// the maximum number of contacts stored in a bucket
-	k = 20
+	// K - the maximum number of contacts stored in a bucket
+	K = 20
 )
 
-// hashTable represents the hashtable state
-type hashTable struct {
+// HashTable represents the hashtable state
+type HashTable struct {
 	// The ID of the local node
-	Self *NetworkNode
+	self *Node
 
-	// Routing table a list of all known nodes in the network
+	// Route table a list of all known nodes in the network
 	// Nodes within buckets are sorted by least recently seen e.g.
 	// [ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ][ ]
 	//  ^                                                           ^
 	//  └ Least recently seen                    Most recently seen ┘
-	RoutingTable [][]*node // 160x20
+	routeTable [][]*Node // 160x20
 
-	mutex *sync.Mutex
+	// mutex for route table
+	mutex sync.Mutex
```

Comment:
can use RWMutex?

Reply:
of course, we can use RLock for the read operation